eidas_node/tests/test_storage.py

```diff
@@ -0,0 +1,100 @@
+from typing import BinaryIO, TextIO, cast
+from unittest.mock import MagicMock, call, patch
+
+from django.test import SimpleTestCase
+
+from eidas_node.models import LightRequest, LightResponse
+from eidas_node.storage.ignite import IgniteStorage
+from eidas_node.tests.test_models import DATA_DIR
+from eidas_node.utils import parse_xml
+
+
+class TestIgniteStorage(SimpleTestCase):
+    HOST = 'localhost.example.net'
+    PORT = 12345
+    REQUEST_CACHE_NAME = 'RequestCacheTest'
+    RESPONSE_CACHE_NAME = 'ResponseCacheTest'
+
+    def setUp(self):
+        self.cache_mock = MagicMock(spec_set=['get', 'put'])
+        self.client_mock = MagicMock(spec_set=['connect', 'get_cache'])
+        self.client_mock.get_cache.return_value = self.cache_mock
+        client_class_patcher = patch('eidas_node.storage.ignite.Client', return_value=self.client_mock)
+        self.client_class_mock = client_class_patcher.start()
+        self.addCleanup(client_class_patcher.stop)
+        self.storage = IgniteStorage(self.HOST, self.PORT, self.REQUEST_CACHE_NAME, self.RESPONSE_CACHE_NAME, 33)
+
+    def test_get_cache_single_client_created(self):
+        self.assertIs(self.storage.get_cache(self.REQUEST_CACHE_NAME), self.cache_mock)
+        self.assertIs(self.storage.get_cache(self.RESPONSE_CACHE_NAME), self.cache_mock)
+        self.assertSequenceEqual(self.client_class_mock.mock_calls, [call(timeout=33)])
```

Comment:
This shouldn't be necessary. `assertEqual` shoud be smart enough...

Reply:
Changed.